server/src/main/java/com/genymobile/scrcpy/Controller.java

```diff
@@ -225,6 +228,65 @@ private boolean injectTouch(int action, long pointerId, Position position, float
             }
         }
 
+        /* If the input device is a mouse:
+         *   - the first button pressed must first generate ACTION_DOWN;
+         *   - all button pressed (including the first one) must generate ACTION_BUTTON_PRESS;
+         *   - all button released (including the last one) must generate ACTION_BUTTON_RELEASE;
+         *   - the last button released must in addition generate ACTION_UP.
+         *
+         * Otherwise, Chrome does not work properly: <https://github.com/Genymobile/scrcpy/issues/3635>
+         */
+        if (source == InputDevice.SOURCE_MOUSE) {
+            int previousButtons = pointer.getButtons();
+            pointer.setButtons(buttons);
+
+            if (action == MotionEvent.ACTION_DOWN) {
+                if (previousButtons == 0) {
```

Comment:
In my v2 branch I tested `buttons == actionButton` to check whether it's the first pressed button, so no need to save previous `buttons` in pointer.

Reply:
Good point, fixed :+1: